packages/odyssey-react/src/components/TextInput/TextInput.test.tsx

```diff
@@ -153,5 +153,25 @@ describe("TextInput", () => {
     expect(ref).toHaveBeenLastCalledWith(screen.getByRole(textBox));
   });
 
+  it("renders the prefix", () => {
+    render(<TextInput label={label} prefix="test prefix" />);
+    const prefixElement = screen.getByText("test prefix");
+    expect(prefixElement).toBeInTheDocument();
+    expect(prefixElement.className).toEqual("prefix");
+  });
+
+  it("renders the suffix", () => {
+    render(<TextInput label={label} suffix="test suffix" />);
+    const suffixElement = screen.getByText("test suffix");
+    expect(suffixElement).toBeInTheDocument();
+    expect(suffixElement.className).toEqual("suffix");
+  });
+
+  it.only("doesn't render the prefix if type is search", () => {
+    render(<TextInput label={label} prefix="test prefix" type="search" />);
+    expect(screen.queryByText("test prefix")).toBeNull();
+    expect(screen.getByLabelText(label)).toHaveAttribute("type", "search");
+  });
```

Comment:
`only` ! We'll want to remove that. I've been meaning to setup a lint rule for this as well!

also, I think we should explore building a helpful union type for this so that developers aren't left wondering why their prefix isn't appearing. Maybe something like this:

```
export interface CommonProps ... {
  type?: "text" | "email" | "url" | "tel" | "password"; // no "search" in union
  ...
}
interface SearchProps extends Omit<CommonProps, "type" | "prefix"> {
  type: "search",
  prefix?: never;
}

export type TextInputProps = CommonProps | SearchProps;

```